blueocean-dashboard/src/main/js/components/karaoke/components/Step.jsx

```diff
@@ -46,33 +41,67 @@ export class Step extends Component {
         const { step } = this.props;
         // needed for running steps as reference
         this.durationInMillis = (this.durationHarmonize(step)).durationInMillis;
-        logger.debug('durationInMillis mounting', this.durationInMillis);
+        log.debug('durationInMillis mounting', this.durationInMillis);
+        this.componentWillReceiveProps(this.props);
     }
+
     /**
      * Mainly implemented due to fetch full log `start=0` for a step
      * @param nextProps
      */
     componentWillReceiveProps(nextProps) {
+        const expanded = this.isFocused(nextProps);
         const nextStart = nextProps.location && nextProps.location.query ? nextProps.location.query.start : undefined;
         const currentStart = this.props.location && this.props.location.query ? this.props.location.query.start : undefined;
-        logger.debug('newProps mate', nextStart, currentStart);
-        if (currentStart !== nextStart && nextStart !== undefined) {
-            logger.debug('re-fetching since result changed and we want to display the full log');
-            this.pager.fetchLog({ url: nextProps.step.logUrl, start: nextStart });
+        log.debug('newProps mate', nextStart, currentStart);
+        if (!this.logStore || (currentStart !== nextStart && nextStart !== undefined)) {
```

Comment:
main issue with this is I somehow broke the logs with the last round of refactoring. *sigh* writing teh codez is hard